src/main/java/org/verapdf/as/io/ASMemoryInStream.java

```diff
@@ -82,13 +83,10 @@ public ASMemoryInStream(InputStream stream) throws IOException {
     public ASMemoryInStream(ASMemoryInStream stream, int offset, int length) {
         this.buffer = stream.buffer;
         this.copiedBuffer = false;
-        if (offset >= 0 && offset < stream.bufferSize) {
-            this.currentPosition = offset;
-        } else {
-            this.currentPosition = 0;
-        }
+        this.bufferOffset = offset + stream.bufferOffset;
```

Comment:
Could you add checks for offset and length limits (similar to checks in getStream method) so that IndexOutOfRangeException does not occur in other methods

Reply:
Like this? https://github.com/veraPDF/veraPDF-parser/compare/d498f551a69899e4c59b44769731277f241a0d0b..29d798ccc0ef6d42df8732aecc0dbfed99798b6e

Reply:
I think should be `if (offset < 0) { ` instead of `if (this.bufferOffset + offset < 0) {`

Reply:
Oh, that was wrong, sorry about that. As for what´s correct, I think maybe it should be `this.bufferOffset`, so that one can have a substream before the start as long as the parent of the stream can handle it: https://github.com/veraPDF/veraPDF-parser/compare/ba542e7c88b799a514cf816d49beb2d1d1b702a4..e856d5f6bf1004dcf51653e72f2bd2be197feca4#diff-a4a186dc0ed9a508badf47413336c24e662829f09b7f0e6ae3f499ab497790b8R206